### PR TITLE
Add .findLast() and .findLastIndex()

### PR DESCRIPTION
**Pull request by m-basov**
https://github.com/microsoft/TypeScript/issues/48829

**Describe your changes**
Add type definitions for `Array.prototype.findLast()` and `Array.prototype.findLastIndex()` methods. Both methods were added to the `es2023` target but the target didn't exist so I've created it. The definitions are identical to `.find()` and `.findIndex()` but with the updated comments.

**Testing performed**
Added `findLast.ts` test file were all Array interfaces try to call `.findLast()` and `.findLastIndex()`. 
Also, tested it manually by running:
- `tsc` with the `es5` target and `libs: []`
- `tsc` with the `es5` target and `libs: ["es2023"]`
- `tsc` with the `es5` target and `libs: ["esnext"]`

**Additional context**
The `es2023` target were added with the help of this PR https://github.com/microsoft/TypeScript/pull/46291

### CODE REVIEW

File: tests/baselines/reference/callChainWithSuper(target=es2023).js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+//// [callChainWithSuper.ts]
```
Comment (mkubilayk) [inline tests/baselines/reference/callChainWithSuper(target=es2023).js:1]: Is this test relevant to your changes?

**Review by mkubilayk — COMMENTED** [review]
Looks great - one small comment.

**Comment by acutmore** [comment]
I wonder if adding es2023 as a target should be separated out as a separate PR. es2023 doesn't exist yet as a spec so it doesn't feel right that TypeScript would offer it as a target yet. 'esnext' seems more appropriate for now.

**Comment by mkubilayk** [comment]
Daniel suggests adding them to ES2023 in https://github.com/microsoft/TypeScript/issues/48829#issuecomment-1147769106.

File: tests/baselines/reference/callChainWithSuper(target=es2023).js (hunk 1 above)
Comment (m-basov) [reply tests/baselines/reference/callChainWithSuper(target=es2023).js:1]: I believe it was created because the `es2023` target was added (filename is `callChainWithSuper(target=es2023).js`). This file [was added](https://github.com/microsoft/TypeScript/pull/46291/files#diff-6bd62e5549898b8187cd7407f28ad8cfc5bd547fac86fb30213002aeaea8c9b3) for the `es2022` target as well.

**Comment by acutmore** [comment]
> Daniel suggests adding them to ES2023 in [microsoft#48829 (comment)](https://github.com/microsoft/TypeScript/issues/48829#issuecomment-1147769106).

Ah cool. Thanks. I can also see that TypeScript also [added es2022 before the spec was official](https://github.com/microsoft/TypeScript/pull/46291/files) so does seem like there is precedent for this. I'm happy.

**Review by dragomirtitian — APPROVED** [review]
